jdaviz/conftest.py

```diff
@@ -179,23 +179,34 @@ def multi_order_spectrum_list(spectrum1d, spectral_orders=10):
     return SpectrumList(sc)
 
 
-def _create_spectrum1d_cube_with_fluxunit(fluxunit=u.Jy, shape=(2, 2, 4)):
+def _create_spectrum1d_cube_with_fluxunit(fluxunit=u.Jy, shape=(2, 2, 4), with_uncerts=False):
 
     flux = np.arange(np.prod(shape)).reshape(shape) * fluxunit
     wcs_dict = {"CTYPE1": "RA---TAN", "CTYPE2": "DEC--TAN", "CTYPE3": "WAVE-LOG",
                 "CRVAL1": 205, "CRVAL2": 27, "CRVAL3": 4.622e-7,
                 "CDELT1": -0.0001, "CDELT2": 0.0001, "CDELT3": 8e-11,
                 "CRPIX1": 0, "CRPIX2": 0, "CRPIX3": 0}
     w = WCS(wcs_dict)
+    if with_uncerts:
+        uncert = StdDevUncertainty(np.abs(np.random.normal(flux) * u.Jy))
 
-    return Spectrum1D(flux=flux, wcs=w)
+        return Spectrum1D(flux=flux,
+                          uncertainty=uncert,
+                          wcs=w)
+    else:
+        return Spectrum1D(flux=flux, wcs=w)
 
 
 @pytest.fixture
 def spectrum1d_cube():
     return _create_spectrum1d_cube_with_fluxunit(fluxunit=u.Jy)
 
 
+@pytest.fixture
+def spectrum1d_cube_with_uncerts():
```

Comment:
turns out a lot of other tests start failing if the existing fixture is updated to include uncertainties (might be worth investigating later... but it looks like at least most of them are caused by assertions on link-counting, etc, which makes sense to fail if the uncertainty cube is available), so I created a new one instead